kernel/src/engine/sync/storage.rs

```diff
@@ -70,6 +70,10 @@ impl StorageHandler for SyncStorageHandler {
         });
         Ok(Box::new(iter))
     }
+
+    fn copy(&self, _src: &Url, _dest: &Url) -> DeltaResult<()> {
+        unimplemented!("SyncStorageHandler does not implement copy");
```

Comment:
hrmm, it's slightly annoying that std doesn't provide a good way to do this. `renameat2` will do what we want, but it's not super easy to expose. do we not need this for testing?